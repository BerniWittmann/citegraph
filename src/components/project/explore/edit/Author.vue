<template>
  <div class="project-explore-single-author-edit">
    <v-card class="pa-4">
      <v-card-title class="headline">
        <h2 class="display-5" v-once>{{ author.firstName }} {{ author.lastName }}</h2>
      </v-card-title>
      <v-container>
        <v-form ref="form" @submit.prevent="submit" class="mt-12" v-model="valid">
          <v-row justify="center">
            <v-col lg="6">
              <v-text-field
                :label="$t('project.explore.edit.author.first_name')"
                v-model="firstName"
                :rules="nameRules"
                required
                autofocus/>
            </v-col>
          </v-row>
          <v-row justify="center">
            <v-col lg="6">
              <v-text-field
                :label="$t('project.explore.edit.author.last_name')"
                v-model="lastName"
                :rules="nameRules"
                required
                autofocus/>
            </v-col>
          </v-row>
          <v-row class="my-10" justify="space-between">
            <v-col cols="auto">
              <v-btn rounded large color="warning" @click="back">
                <v-icon>mdi-menu-left</v-icon>
                {{ $t('project.explore.edit.back') }}
              </v-btn>
            </v-col>

            <v-col cols="auto">
              <v-btn
                rounded large
                :disabled="!valid"
                :loading="loading"
                color="success"
                type="submit"
              >{{ $t('project.explore.edit.submit') }}
              </v-btn>
            </v-col>
          </v-row>
        </v-form>
      </v-container>
    </v-card>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import Author from '@/models/paperEntities/author'

@Component
export default class AuthorEditPage extends Vue {
  valid: boolean = true
  loading: boolean = false
  firstName: string = ''
  lastName: string = ''

  nameRules: Array<Function> = [
    (v: string | undefined) => !!v || this.$root.$t('project.explore.edit.author.name_required'),
    (v: string | undefined) => (v && v.trim().length > 0) || this.$root.$t('project.explore.edit.author.name_not_empty')
  ]

  urlRules: Array<Function> = []

  get author (): Author {
    return this.$store.getters['paperEntities/activeEntity'] as Author
  }

  beforeMount (): void {
    this.firstName = this.author.firstName
    this.lastName = this.author.lastName
  }

  async submit (): Promise<void> {
    if (!this.valid) return
    this.loading = true
    await this.$store.dispatch('paperEntities/updateEntity', {
      params: {
        projectId: this.$route.params.projectId,
        entityType: this.$route.params.queryByType,
        id: this.$route.params.entityId
      },
      data: {
        firstName: this.firstName,
        lastName: this.lastName
      }
    })
    this.loading = false
    this.$router.back()
  }

  back (): void {
    this.$router.back()
  }
}
</script>
