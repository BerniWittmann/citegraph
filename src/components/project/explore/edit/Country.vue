<template>
  <div class="project-explore-single-country-edit">
    <v-card class="pa-4">
      <v-card-title class="headline">
        <h2 class="display-5" v-once>{{country.name }}</h2>
      </v-card-title>
      <v-container>
        <v-form ref="form" @submit.prevent="submit" class="mt-12" v-model="valid">
          <v-row justify="center">
            <v-col lg="6">
              <v-text-field
                :label="$t('project.explore.edit.country.name')"
                v-model="name"
                :rules="nameRules"
                required
                autofocus/>
            </v-col>
          </v-row>
          <v-row justify="center">
            <v-col lg="6">
              <v-text-field
                :label="$t('project.explore.edit.country.flag_url')"
                v-model="flagUrl"
                :rules="urlRules"
                >
                <template v-slot:append>
                  <v-img v-if="flagUrl" :src="flagUrl" :lazy-src="flagUrl" max-width="30" width="30" contain>
                    <template v-slot:placeholder>
                      <v-row
                        class="fill-height ma-0"
                        align="center"
                        justify="center"
                      >
                        <v-progress-circular indeterminate color="grey lighten-5"/>
                      </v-row>
                    </template>
                  </v-img>
                </template>
              </v-text-field>
            </v-col>
          </v-row>
          <v-row class="my-10" justify="space-between">
            <v-col cols="auto">
              <v-btn rounded large color="warning" @click="back">
                <v-icon>mdi-menu-left</v-icon>
                {{ $t('project.explore.edit.back') }}
              </v-btn>
            </v-col>

            <v-col cols="auto">
              <v-btn
                rounded large
                :disabled="!valid"
                :loading="loading"
                color="success"
                type="submit"
              >{{ $t('project.explore.edit.submit') }}
              </v-btn>
            </v-col>
          </v-row>
        </v-form>
      </v-container>
    </v-card>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import Country from '@/models/paperEntities/country'
import DataTable from '@/components/project/explore/DataTable.vue'

@Component({
  components: {
    DataTable
  }
})
export default class CountryEditPage extends Vue {
  valid: boolean = true
  loading: boolean = false
  name: string = ''
  flagUrl?: string

  nameRules: Array<Function> = [
    (v: string | undefined) => !!v || this.$root.$t('project.explore.edit.country.name_required'),
    (v: string | undefined) => (v && v.trim().length > 0) || this.$root.$t('project.explore.edit.country.name_not_empty')
  ]

  urlRules: Array<Function> = []

  get country (): Country {
    return this.$store.getters['paperEntities/activeEntity'] as Country
  }

  beforeMount (): void {
    this.name = this.country.name
    this.flagUrl = this.country.flagUrl
  }

  async submit (): Promise<void> {
    if (!this.valid) return
    this.loading = true
    await this.$store.dispatch('paperEntities/updateEntity', {
      params: {
        projectId: this.$route.params.projectId,
        entityType: this.$route.params.queryByType,
        id: this.$route.params.entityId
      },
      data: {
        name: this.name,
        flagUrl: this.flagUrl
      }
    })
    this.loading = false
    this.$router.back()
  }

  back (): void {
    this.$router.back()
  }
}
</script>
