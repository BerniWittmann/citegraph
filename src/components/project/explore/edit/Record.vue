<template>
  <div class="project-explore-single-record-edit">
    <v-card class="pa-4">
      <v-card-title class="headline">
        <h2 class="display-5" v-once>{{record.title }}</h2>
      </v-card-title>
      <v-container>
        <v-form ref="form" @submit.prevent="submit" class="mt-12" v-model="valid">
          <v-row justify="center">
            <v-col lg="6">
              <v-text-field
                :label="$t('project.explore.edit.record.title')"
                v-model="title"
                :rules="nameRules"
                required
                autofocus/>
            </v-col>
          </v-row>
          <v-row class="my-10" justify="space-between">
            <v-col cols="auto">
              <v-btn rounded large color="warning" @click="back">
                <v-icon>mdi-menu-left</v-icon>
                {{ $t('project.explore.edit.back') }}
              </v-btn>
            </v-col>

            <v-col cols="auto">
              <v-btn
                rounded large
                :disabled="!valid"
                :loading="loading"
                color="success"
                type="submit"
              >{{ $t('project.explore.edit.submit') }}
              </v-btn>
            </v-col>
          </v-row>
        </v-form>
      </v-container>
    </v-card>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import Record from '@/models/paperEntities/record'

@Component
export default class RecordEditPage extends Vue {
  valid: boolean = true
  loading: boolean = false
  title: string = ''

  nameRules: Array<Function> = [
    (v: string | undefined) => !!v || this.$root.$t('project.explore.edit.record.name_required'),
    (v: string | undefined) => (v && v.trim().length > 0) || this.$root.$t('project.explore.edit.record.name_not_empty')
  ]

  get record (): Record {
    return this.$store.getters['paperEntities/activeEntity'] as Record
  }

  beforeMount (): void {
    this.title = this.record.title
  }

  async submit (): Promise<void> {
    if (!this.valid) return
    this.loading = true
    await this.$store.dispatch('paperEntities/updateEntity', {
      params: {
        projectId: this.$route.params.projectId,
        entityType: this.$route.params.queryByType,
        id: this.$route.params.entityId
      },
      data: {
        title: this.title
      }
    })
    this.loading = false
    this.$router.back()
  }

  back (): void {
    this.$router.back()
  }
}
</script>
