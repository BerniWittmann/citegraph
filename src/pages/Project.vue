<template>
  <div class="project">
    <v-alert v-if="!hasProject" type="error" prominent>
      {{ $t('project.not_available') }}
    </v-alert>

    <div v-if="hasProject">
      <h1>{{ project.name }}</h1>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import Project from '@/models/project'

@Component
export default class ProjectPage extends Vue {
  get project (): Project | undefined {
    return this.$store.getters['projects/activeProject']
  }

  get hasProject (): boolean {
    return this.$store.getters['projects/hasActiveProject']
  }
}
</script>
