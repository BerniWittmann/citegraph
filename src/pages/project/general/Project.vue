import { ProjectStates } from '@/models/project'
<template>
  <div class="project">
    <div class="d-flex mr-5 mt-4">
      <h1>{{ project.name }}</h1>
      <v-spacer/>
      <v-btn :to="{ name: 'projects.single.edit', params: { projectId: project.id } }">
        <v-icon>mdi-pencil</v-icon>
        {{ $t('project.edit.button_text') }}
      </v-btn>
    </div>
    <v-container v-if="hasNoDataYet">
      <v-row justify="center">
        <v-col :lg="3" class="d-flex justify-center flex-column text-center">
          <empty-icon class="align-self-center mb-6"></empty-icon>
          <h3 class="headline">{{ $t('project.single.no_data.description') }}</h3>
          <p>{{ $t('project.single.no_data.import_data') }}</p>
          <v-btn color="primary" rounded large :to="{ name: 'projects.single.import', params: { projectId: project.id } }">
            {{ $t('project.single.no_data.button') }}
          </v-btn>
        </v-col>
      </v-row>
    </v-container>
    <v-container v-else>
      <v-row v-if="datasetInformation">
        <v-col>
          <v-card>
            <v-card-title>{{ $t('project.single.dataset.title') }}</v-card-title>
            <v-card-text>
              <v-container>
                <v-row>
                  <v-col v-for="(count, name) in datasetInformation" :key="name">
                    <v-card outlined class="d-flex flex-column justify-center text-center pa-5">
                      <p class="headline">{{count}}</p>
                      <p class="mb-0">{{ $t('project.single.dataset.' + name ) }}</p>
                    </v-card>
                  </v-col>
                </v-row>
              </v-container>
            </v-card-text>
          </v-card>
        </v-col>
      </v-row>
      <v-row>
        <v-col lg="3" md="6" cols="12" v-if="project.topAuthors">
          <v-card>
            <v-card-title>
              {{ $t('project.single.overview.top_10_authors') }}
            </v-card-title>
            <v-card-text>
              <v-list>
                <v-list-item v-for="author in project.topAuthors" :key="author.id" :to="{ name: 'projects.single.explore.view', params: {
                 projectId: project.id,
                 queryByType: 'author',
                 entityId: author.id
                }}">
                  {{ author.firstName }} {{ author.lastName }}
                </v-list-item>
              </v-list>
            </v-card-text>
          </v-card>
        </v-col>
        <v-col lg="3" md="6" cols="12" v-if="project.topRecords">
          <v-card>
            <v-card-title>
              {{ $t('project.single.overview.top_10_records') }}
            </v-card-title>
            <v-card-text>
              <v-list>
                <v-list-item v-for="record in project.topRecords" :key="record.id" :to="{ name: 'projects.single.explore.view', params: {
                 projectId: project.id,
                 queryByType: 'record',
                 entityId: record.id
                }}">
                  {{ record.title }}
                </v-list-item>
              </v-list>
            </v-card-text>
          </v-card>
        </v-col>
        <v-col lg="3" md="6" cols="12" v-if="project.countryRelevanceData">
          <v-card>
            <v-card-title>
              {{ $t('project.single.overview.country_relevance') }}
            </v-card-title>
            <v-card-text>
              TODO Diagram here
            </v-card-text>
          </v-card>
        </v-col>
        <v-col lg="3" md="6" cols="12">
          <v-row no-gutters>
            <v-col v-if="project.topKeywords" class="mb-6">
              <v-card>
                <v-card-title>
                  {{ $t('project.single.overview.top_10_keywords') }}
                </v-card-title>
                <v-card-text>
                  <v-chip-group multiple column>
                    <v-chip v-for="keyword in project.topKeywords" :key="keyword">{{ keyword }}</v-chip>
                  </v-chip-group>
                </v-card-text>
              </v-card>
            </v-col>
          </v-row>
          <v-row no-gutters>
            <v-col v-if="project.counts">
              <v-card>
                <v-card-title>
                  {{ $t('project.single.overview.visualizations') }}
                </v-card-title>
                <v-card-text>
                  <div class="d-flex flex-column justify-center text-center pa-5" v-if="hasNoVisualizationsYet">
                    <p class="mb-0">{{ $t('project.single.overview.no_visualizations_yet') }}</p>
                    <v-btn color="primary" rounded>{{ $t('project.single.overview.add_visualization') }}</v-btn>
                  </div>
                  <v-card outlined class="d-flex flex-column justify-center text-center pa-5" v-else>
                    <p class="headline">{{ project.counts.visualizations }}</p>
                    <p class="mb-0">{{ $t('project.single.overview.visualizations') }}</p>
                  </v-card>
                </v-card-text>
              </v-card>
            </v-col>
          </v-row>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import Project, { ProjectStates } from '@/models/project'
import EmptyIcon from '@/assets/svgs/Empty.vue'

@Component({
  components: {
    EmptyIcon
  }
})
export default class ProjectPage extends Vue {
  get project (): Project {
    return this.$store.getters['projects/activeProject']
  }

  get datasetInformation (): Object | undefined {
    if (!this.project.counts) return undefined
    const data = { ...this.project.counts }
    delete data['visualizations']
    return data
  }

  get hasNoDataYet (): boolean {
    return this.project.state === ProjectStates.IMPORT
  }

  get hasNoVisualizationsYet (): boolean {
    return this.project.state === ProjectStates.EXPLORE
  }
}
</script>
