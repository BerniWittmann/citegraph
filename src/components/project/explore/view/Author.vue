<template>
  <div class="project-explore-single-author" v-if="author">
    <layout>
      <template v-slot:title>
        <v-avatar color="primary" class="mr-5" size="62">
          <b class="white--text">{{ initials }}</b>
        </v-avatar>
        <h2 class="display-5">{{ author.firstName }} {{ author.lastName }}</h2>
      </template>
      <v-container>
        <v-row>
          <v-col :xl="3" :lg="6">
            <v-list>
              <v-subheader class="title">{{ $t('project.explore.view.author.metrics') }}</v-subheader>
              <v-list-item>
                <v-list-item-content>{{ $t('project.explore.view.author.count_records') }}:</v-list-item-content>
                <v-list-item-content class="align-end">{{ author.countRecords }}</v-list-item-content>
              </v-list-item>
            </v-list>
          </v-col>
        </v-row>
        <v-row>
          <data-table belongs-to-type="author" :belongs-to="author.id" query-by-type="record"></data-table>
        </v-row>
      </v-container>
    </layout>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import PaperEntityLayout from '@/layouts/PaperEntity.vue'
import Author from '@/models/paperEntities/author'
import DataTable from '@/components/project/explore/DataTable.vue'

@Component({
  components: {
    DataTable,
    Layout: PaperEntityLayout
  }
})
export default class AuthorViewPage extends Vue {
  get author (): Author {
    return this.$store.getters['paperEntities/activeEntity'] as Author
  }

  get initials (): string {
    return ((this.author.firstName ? this.author.firstName.substring(0, 1) : '') + (this.author.lastName ? this.author.lastName.substring(0, 1) : '')).toUpperCase()
  }
}
</script>
