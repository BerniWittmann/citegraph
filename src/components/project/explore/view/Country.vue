<template>
  <div class="project-explore-single-country">
    <layout>
      <template v-slot:title>
        <v-img v-if="country.flagUrl" :src="country.flagUrl" max-width="60" width="60" contain></v-img>
        <v-sheet v-else color="grey lighten-1" class="text-center" width="60">
          <v-icon dark>mdi-help</v-icon>
        </v-sheet>
        <h2 class="display-5 ml-5">{{country.name }}</h2>
      </template>
      <v-container>
        <v-row>
          <v-col :xl="3" :lg="6">
            <v-list>
              <v-subheader class="title">{{ $t('project.explore.view.country.metrics') }}</v-subheader>
              <v-list-item>
                <v-list-item-content>{{ $t('project.explore.view.country.count_records') }}:</v-list-item-content>
                <v-list-item-content class="align-end">{{ country.countRecords }}</v-list-item-content>
              </v-list-item>
            </v-list>
          </v-col>
        </v-row>
      </v-container>
    </layout>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import PaperEntityLayout from '@/layouts/PaperEntity.vue'
import Country from '@/models/paperEntities/country'

@Component({
  components: {
    Layout: PaperEntityLayout
  }
})
export default class CountryViewPage extends Vue {
  get country (): Country {
    return this.$store.getters['paperEntities/activeEntity'] as Country
  }
}
</script>
