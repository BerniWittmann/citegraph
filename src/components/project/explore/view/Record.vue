<template>
  <div class="project-explore-single-record">
    <layout>
      <template v-slot:title>
        <h2 class="display-5">{{ record.title }}</h2>
      </template>
      <v-container>
        <v-row>
          <v-col>
            <v-subheader class="title">{{ $t('project.explore.view.record.authors') }}</v-subheader>
            <v-chip-group column>
              <v-chip v-for="(author, index) in record.authors" :key="index"
                      :to="{ name: 'projects.single.explore.view', params: { ...$route.params, queryByType: 'author', entityId: author.id } }">
                {{ author.firstName }} {{ author.lastName }}
              </v-chip>
            </v-chip-group>
          </v-col>
          <v-col :cols="2">
            <v-card class="justify-center text-center pa-2 py-3">
              <b class="display-1 primary--text font-weight-black">{{ record.numberCitations }}</b>
              <p class="headline">{{ $t('project.explore.view.record.citations') }}</p>
            </v-card>
          </v-col>
        </v-row>
        <v-row>
          <v-col>
            <v-subheader class="title">{{ $t('project.explore.view.record.year') }}</v-subheader>
            <p>{{ record.year }}</p>
          </v-col>
        </v-row>
        <v-row>
          <v-col>
            <v-subheader class="title">{{ $t('project.explore.view.record.keywords') }}</v-subheader>
            <v-chip-group column>
              <v-chip v-for="(keyword, index) in record.keywords" small :key="index">{{ keyword }}</v-chip>
            </v-chip-group>
          </v-col>
        </v-row>
      </v-container>
    </layout>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import PaperEntityLayout from '@/layouts/PaperEntity.vue'
import Record from '@/models/paperEntities/record'

@Component({
  components: {
    Layout: PaperEntityLayout
  }
})
export default class RecordViewPage extends Vue {
  get record (): Record {
    return this.$store.getters['paperEntities/activeEntity'] as Record
  }
}
</script>
