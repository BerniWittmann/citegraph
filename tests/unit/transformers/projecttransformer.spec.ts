import ProjectTransformer from '@/transformers/ProjectTransformer'
import Project from '@/models/project'
import Author from '@/models/paperEntities/author'
import Record from '@/models/paperEntities/record'
import Country from '@/models/paperEntities/country'

describe('transformers/ProjectTransformer', () => {
  describe('can send a project', () => {
    it('sends a project', () => {
      expect(ProjectTransformer.send(new Project({ name: 'Project' }))).toMatchSnapshot()
    })

    it('sends the counts', () => {
      expect(ProjectTransformer.send(new Project({
        name: 'Project',
        counts: {
          records: 1,
          authors: 2,
          institutions: 3,
          countries: 4,
          publications: 5,
          publishers: 6,
          conferences: 7,
          keywords: 8,
          visualizations: 9
        }
      }))).toMatchSnapshot()
    })

    it('sends the authors', () => {
      expect(ProjectTransformer.send(new Project({
        name: 'Project',
        topAuthors: [
          new Author({ id: '1', firstName: 'First', lastName: 'Author', countRecords: 1 }),
          new Author({ id: '2', firstName: 'Second', lastName: 'Author', countRecords: 2 })
        ]
      }))).toMatchSnapshot()
    })

    it('sends the records', () => {
      expect(ProjectTransformer.send(new Project({
        name: 'Project',
        topRecords: [
          new Record({ id: '1', title: 'First', year: 2000, keywords: [], authors: [], numberCitations: 1 }),
          new Record({ id: '2', title: 'Second', year: 2000, keywords: [], authors: [], numberCitations: 1 })
        ]
      }))).toMatchSnapshot()
    })

    it('sends the country relevance data', () => {
      expect(ProjectTransformer.send(new Project({
        name: 'Project',
        countryRelevanceData: [
          {
            country: new Country({ id: '1', name: 'Deutschland', countRecords: 1 }),
            relevance: 0.4
          }
        ]
      }))).toMatchSnapshot()
    })
  })

  describe('can fetch a project', () => {
    it('fetches a project', () => {
      expect(ProjectTransformer.fetch({ name: 'Project' })).toMatchSnapshot()
    })

    it('fetches the counts', () => {
      expect(ProjectTransformer.fetch({
        name: 'Project',
        count_records: 1,
        count_authors: 2,
        count_institutions: 3,
        count_countries: 4,
        count_publications: 5,
        count_publishers: 6,
        count_conferences: 7,
        count_keywords: 8,
        count_visualizations: 9
      })).toMatchSnapshot()
    })

    it('fetches the authors', () => {
      expect(ProjectTransformer.fetch({
        name: 'Project',
        top_authors: [
          { id: '1', first_name: 'First', last_name: 'Author', count_records: 1 },
          { id: '2', first_name: 'Second', last_name: 'Author', count_records: 2 }
        ]
      })).toMatchSnapshot()
    })

    it('fetches the records', () => {
      expect(ProjectTransformer.fetch({
        name: 'Project',
        top_records: [
          { id: '1', title: 'First', year: 2000, keywords: [], authors: [], number_citations: 1 },
          { id: '2', title: 'Second', year: 2000, keywords: [], authors: [], number_citations: 1 }
        ]
      })).toMatchSnapshot()
    })

    it('fetches the country relevance data', () => {
      expect(ProjectTransformer.fetch({
        name: 'Project',
        countries: [
          {
            id: '1',
            name: 'Deutschland',
            count_records: 1,
            relevance: 0.4
          }
        ]
      })).toMatchSnapshot()
    })
  })
})
