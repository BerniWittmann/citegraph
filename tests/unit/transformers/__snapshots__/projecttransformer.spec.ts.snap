// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformers/ProjectTransformer can fetch a project fetches a project 1`] = `
Project {
  "countryRelevanceData": undefined,
  "counts": undefined,
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "topAuthors": undefined,
  "topKeywords": undefined,
  "topRecords": undefined,
}
`;

exports[`transformers/ProjectTransformer can fetch a project fetches the authors 1`] = `
Project {
  "countryRelevanceData": undefined,
  "counts": undefined,
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "topAuthors": Array [
    Author {
      "countRecords": 1,
      "firstName": "First",
      "id": "1",
      "lastName": "Author",
    },
    Author {
      "countRecords": 2,
      "firstName": "Second",
      "id": "2",
      "lastName": "Author",
    },
  ],
  "topKeywords": undefined,
  "topRecords": undefined,
}
`;

exports[`transformers/ProjectTransformer can fetch a project fetches the country relevance data 1`] = `
Project {
  "countryRelevanceData": Array [
    Object {
      "country": Country {
        "countRecords": 1,
        "flagUrl": undefined,
        "id": "1",
        "name": "Deutschland",
      },
      "relevance": 0.4,
    },
  ],
  "counts": undefined,
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "topAuthors": undefined,
  "topKeywords": undefined,
  "topRecords": undefined,
}
`;

exports[`transformers/ProjectTransformer can fetch a project fetches the counts 1`] = `
Project {
  "countryRelevanceData": undefined,
  "counts": Object {
    "authors": 2,
    "conferences": 7,
    "countries": 4,
    "institutions": 3,
    "keywords": 8,
    "publications": 5,
    "publishers": 5,
    "records": 1,
    "visualizations": 9,
  },
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "topAuthors": undefined,
  "topKeywords": undefined,
  "topRecords": undefined,
}
`;

exports[`transformers/ProjectTransformer can fetch a project fetches the records 1`] = `
Project {
  "countryRelevanceData": undefined,
  "counts": undefined,
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "topAuthors": undefined,
  "topKeywords": undefined,
  "topRecords": Array [
    Record {
      "authors": Array [],
      "id": "1",
      "keywords": Array [],
      "numberCitations": 1,
      "title": "First",
      "year": 2000,
    },
    Record {
      "authors": Array [],
      "id": "2",
      "keywords": Array [],
      "numberCitations": 1,
      "title": "Second",
      "year": 2000,
    },
  ],
}
`;

exports[`transformers/ProjectTransformer can send a project sends a project 1`] = `
Object {
  "count_authors": undefined,
  "count_conferences": undefined,
  "count_countries": undefined,
  "count_institutions": undefined,
  "count_keywords": undefined,
  "count_publications": undefined,
  "count_publishers": undefined,
  "count_records": undefined,
  "count_visualizations": undefined,
  "countries": undefined,
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "top_authors": undefined,
  "top_keywords": undefined,
  "top_records": undefined,
}
`;

exports[`transformers/ProjectTransformer can send a project sends the authors 1`] = `
Object {
  "count_authors": undefined,
  "count_conferences": undefined,
  "count_countries": undefined,
  "count_institutions": undefined,
  "count_keywords": undefined,
  "count_publications": undefined,
  "count_publishers": undefined,
  "count_records": undefined,
  "count_visualizations": undefined,
  "countries": undefined,
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "top_authors": Array [
    Object {
      "count_records": 1,
      "first_name": "First",
      "id": "1",
      "last_name": "Author",
    },
    Object {
      "count_records": 2,
      "first_name": "Second",
      "id": "2",
      "last_name": "Author",
    },
  ],
  "top_keywords": undefined,
  "top_records": undefined,
}
`;

exports[`transformers/ProjectTransformer can send a project sends the country relevance data 1`] = `
Object {
  "count_authors": undefined,
  "count_conferences": undefined,
  "count_countries": undefined,
  "count_institutions": undefined,
  "count_keywords": undefined,
  "count_publications": undefined,
  "count_publishers": undefined,
  "count_records": undefined,
  "count_visualizations": undefined,
  "countries": Array [
    Object {
      "count_records": 1,
      "flag_url": undefined,
      "id": "1",
      "name": "Deutschland",
      "relevance": 0.4,
    },
  ],
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "top_authors": undefined,
  "top_keywords": undefined,
  "top_records": undefined,
}
`;

exports[`transformers/ProjectTransformer can send a project sends the counts 1`] = `
Object {
  "count_authors": 2,
  "count_conferences": 7,
  "count_countries": 4,
  "count_institutions": 3,
  "count_keywords": 8,
  "count_publications": 5,
  "count_publishers": 6,
  "count_records": 1,
  "count_visualizations": 9,
  "countries": undefined,
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "top_authors": undefined,
  "top_keywords": undefined,
  "top_records": undefined,
}
`;

exports[`transformers/ProjectTransformer can send a project sends the records 1`] = `
Object {
  "count_authors": undefined,
  "count_conferences": undefined,
  "count_countries": undefined,
  "count_institutions": undefined,
  "count_keywords": undefined,
  "count_publications": undefined,
  "count_publishers": undefined,
  "count_records": undefined,
  "count_visualizations": undefined,
  "countries": undefined,
  "id": undefined,
  "name": "Project",
  "state": "IMPORT",
  "top_authors": undefined,
  "top_keywords": undefined,
  "top_records": Array [
    Object {
      "authors": Array [],
      "id": "1",
      "keywords": Array [],
      "number_citations": 1,
      "title": "First",
      "year": 2000,
    },
    Object {
      "authors": Array [],
      "id": "2",
      "keywords": Array [],
      "number_citations": 1,
      "title": "Second",
      "year": 2000,
    },
  ],
}
`;
